### PR TITLE
Fix incompatible pointer warnings on macOS with GCC 14

### PR DESCRIPTION
**Pull request by raffenet**
Fixes compilation issue reported in #10052.

### CODE REVIEW

File: src/iov.c [hunk 1]
```diff
@@ -37,8 +37,8 @@
 #include <ofi.h>
 #include <ofi_iov.h>
 
-uint64_t ofi_copy_iov_buf(const struct iovec *iov, size_t iov_count, uint64_t iov_offset,
-			  void *buf, uint64_t bufsize, int dir)
+uint64_t ofi_copy_iov_buf(const struct iovec *iov, size_t iov_count, size_t iov_offset,
```
Comment (darrylabbate) [inline src/iov.c:40]: Probably a good idea to change the return type as well. I wasn't able to find any context wrt why `uint64_t` was chosen over `size_t`, but it's likely an oversight considering the compatibility.

Comment (raffenet) [reply src/iov.c:40]: Makes sense to me. Let me give it a try.

**Review by darrylabbate — APPROVED** [review]

File: src/iov.c [hunk 2]
```diff
@@ -37,8 +37,8 @@
 #include <ofi.h>
 #include <ofi_iov.h>
 
-uint64_t ofi_copy_iov_buf(const struct iovec *iov, size_t iov_count, uint64_t iov_offset,
-			  void *buf, uint64_t bufsize, int dir)
+size_t ofi_copy_iov_buf(const struct iovec *iov, size_t iov_count, size_t iov_offset,
+                        void *buf, size_t bufsize, int dir)
 {
 	uint64_t done = 0, len;
```
Comment (darrylabbate) [inline src/iov.c:43]: Change to `size_t` (sorry; missed this).

Comment (raffenet) [reply src/iov.c:43]: No worries. New version coming.

Comment (raffenet) [reply src/iov.c]: @darrylabbate I added one more commit to convert the remaining `uint64_t` in `src/iov.c`. I can squash before merge if it looks OK.

Comment (darrylabbate) [reply src/iov.c]: I think this LGTM. Much appreciated.

Comment (raffenet) [reply src/iov.c]: 👍 should I squash the first and third commits? Anything else to do before merging?

Comment (darrylabbate) [reply src/iov.c]: You can squash the first and third commits. If no one else has comments, we can merge after CI passes.

**Pull request merged**